en/getting-started/first-game/index.md

```diff
@@ -370,173 +832,203 @@ export class GameManager extends Component {
+- GS_PLAYING(play): Hide the game menu, and the user can start to control the character.
+- GS_END(end): End the game and show the start menu.
 
-    ```ts
-    private _curState: GameState = GameState.GS_INIT;
-    ```
+The current game state can be easily informed by reading the `GameState` enumeration, and we can handle some logic when the state changes.
 
-    In-order not to let the user operate the character at the beginning, but to allow the user to operate the character while the game is in progress, we need to dynamically turn on and off the character's monitoring of mouse messages. This can be done with the following changes to `PlayerController`:
+In order to allow the user to control the character in the play state rather than the game initialize, we have to enable/disable listening to the mouse automatically. So modify the `PlayerController` script as follows:
 
-    ```ts
-    start () {
-        // Your initialization goes here.
-        //input.on(Input.EventType.MOUSE_UP, this.onMouseUp, this);
-    }
 
-    setInputActive(active: boolean) {
-        if (active) {
-            input.on(Input.EventType.MOUSE_UP, this.onMouseUp, this);
-        } else {
-            input.off(Input.EventType.MOUSE_UP, this.onMouseUp, this);
-        }
+```ts
+start () {
+    // Your initialization goes here.
+    // input.on(Input.EventType.MOUSE_UP, this.onMouseUp, this);
+}
+
+setInputActive(active: boolean) {
+    if (active) {
+        input.on(Input.EventType.MOUSE_UP, this.onMouseUp, this);
+    } else {
+        input.off(Input.EventType.MOUSE_UP, this.onMouseUp, this);
     }
-    ```
+}
+```
 
-    Next, reference `PlayerController` in the `GameManager` script. Drag the `Player` variable in the __Inspector__ panel.
+And reference the `PlayerController` script in the `GameManager` script.
 
-    ```ts
-    @property({type: PlayerController})
-    public playerCtrl: PlayerController = null;
-    ```
+```ts
+@property({type: PlayerController})
+public playerCtrl: PlayerController | null = null;
+```
 
-    In-order to dynamically open/close the open menu, the `StartMenu` needs to be referenced in the `GameManager`. Drag the `StartMenu` of the scene into this variable in the __Inspector__ panel.
+Back to the editor after saving the scripts, and drag the Player node that attached a `PlayerController` script to the `playerCtrl` property of the GameManager Node on the **Inspector** panel.
 
-    ```ts
-    @property({type: Node})
-    public startMenu: Node = null;
-    ```
+Meanwhile, we need to reference the StartMenu node in the `GameManager` script to dynamically enable/disable the start menu.
 
-    ![add player to game manager](./images/game-manager-player.png)
+```ts
+@property({type: Node})
+public startMenu: Node | null = null;
+```
 
-    Modify the code in the `GameManager`:
+Return to the editor after saving all scripts and drag the StartMenu node to the `startMenu` property of the GameManager node on the **Inspector** panel.
```

Comment:
consider carefully about the prepositions:  drag "xxx" node on the "xxx" property in the "XXXXX" panel